packages/vaadin-icon/src/vaadin-iconset.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+import { ElementMixin } from '@vaadin/vaadin-element-mixin/vaadin-element-mixin.js';
+import { nothing, svg } from 'lit';
+import { unsafeSVG } from 'lit/directives/unsafe-svg.js';
+
+const iconRegistry = {};
+
+export const getIconId = (id) => (id || '').replace('vaadin-icon:', '');
+
+/**
+ * `<vaadin-iconset>` is a Web Component for creating SVG icon collections.
+ *
+ * @extends HTMLElement
+ * @mixes ElementMixin
+ */
+class IconsetElement extends ElementMixin(PolymerElement) {
+  static get template() {
+    return null;
+  }
+
+  static get is() {
+    return 'vaadin-iconset';
+  }
+
+  static get properties() {
+    return {
+      /**
+       * The name of the iconset.
+       */
+      name: {
+        type: String,
+        observer: '__nameChanged'
+      }
+    };
+  }
+
+  /**
+   * Create an instance of the iconset.
+   *
+   * @param {string} name
+   */
+  static getIconset(name) {
+    let iconset = iconRegistry[name];
+    if (!iconset) {
+      iconset = document.createElement('vaadin-iconset');
+      iconset.name = name;
+      iconRegistry[name] = iconset;
+    }
+    return iconset;
+  }
+
+  /** @protected */
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.style.display = 'none';
+  }
+
+  /**
+   * Produce SVGTemplateResult for the element matching `id` in this
+   * iconset, or `undefined` if there is no matching element.
+   *
+   * @param {string} id
+   */
+  applyIcon(id) {
+    // create the icon map on-demand, since the iconset itself has no discrete
+    // signal to know when it's children are fully parsed
+    this._icons = this._icons || this.__createIconMap();
+    return this.__prepareSvg(this._icons[id]);
+  }
+
+  /**
+   * Create a map of child SVG elements by id.
+   */
+  __createIconMap() {
+    const icons = Object.create(null);
+    this.querySelectorAll('[id]').forEach((icon) => {
+      icons[getIconId(icon.id)] = icon;
+    });
+    return icons;
+  }
+
+  /** @private */
+  __prepareSvg(sourceSvg) {
+    let result = nothing;
+    if (sourceSvg) {
+      const content = sourceSvg.cloneNode(true);
+      content.removeAttribute('id');
+      result = svg`${unsafeSVG(content.innerHTML)}`;
```

Comment:
I'm not really happy about this approach. We could consider some basic sanitization - [example](https://github.com/mattkrick/sanitize-svg/blob/51ca0db1fdcccb25c18496cf4724813edef3fb19/src/sanitizeSVG.ts#L24-L30).